Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -416,42 +443,15 @@ Investigate the following JWT vulnerabilities while performing dynamic analysis:
   - Token signatures are created via a private key on the server. After you obtain a JWT, choose a tool for [brute forcing the secret key offline](https://www.sjoerdlangkemper.nl/2016/09/28/attacking-jwt-authentication/ "Attacking JWT Authentication").
 - Information Disclosure:
   - Decode the Base64-encoded JWT and find out what kind of data it transmits and whether that data is encrypted.
+- Tampering with the Hashing Algorithm:
+  - Modify the `alg` attribute in the token header, then delete `HS256`, set it to `none`, and use an empty signature (e.g., signature = ""). Use this token and replay it in a request. Some libraries treat tokens signed with the none algorithm as a valid token with a verified signature. This allows attackers to create their own "signed" tokens.
```

Comment:
+ Add the async encryption <-> sync encryption attack

Example here: https://medium.com/101-writeups/hacking-json-web-token-jwt-233fe6c862e6 (but there's probably a better reference)

Reply:
This is mentioned by the first point of this list, referencing to https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/. But I moved it now here as it fits better here, you are right!